### PR TITLE
Bug 1385363: add /v1/oidc-credentials endpoint

### PR DESCRIPTION
**Pull request by djmitche**
So far a work in progress.  @eliperelman what do you think?

The main thing I need to do now is figure out how to interpret the profile..

### CODE REVIEW

**Comment by eliperelman** [comment]
I'm confused as to how this is different from the previous proof-of-concept. Also, didn't we want to create something standalone to demo?

**Comment by djmitche** [comment]
The difference is in the API method.  The TEMPORARY stuff is temporary, yeah, and we want something standalone intsead.  I just copied it from the proof of concept so I could test it out.

**Comment by djmitche** [comment]
@jonasfj I'd like to get this merged now.  There's work to do in giving better scopes, and setting expiration, etc., but that can come in subsequent revisions.  I suspect this will co-evolve as the IAM team makes changes and as @eliperelman develops the demo

File: src/handlers/mozilla-auth0.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  // Get a management API instance, by requesting an API token as needed
+  // see https://auth0.com/docs/api/management/v2/tokens
+  async getManagementApi() {
+    if (this._managementApi && new Date().getTime() / 1000 < this._managementApiExp - 10) {
+      return this._managementApi;
+    }
+
+    let res = await request.post(`https://${this.domain}/oauth/token`)
```
Comment (jonasfj) [inline src/handlers/mozilla-auth0.js]: I can't remember superagent-promise anymore... but make sure this does retries.
(maybe consider using `got`)

File: src/handlers/mozilla-auth0.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    } catch(err) {
+      debug(`error validating jwt: ${err}`);
+      return;
+    }
+
+    debug(`received valid access_token for subject ${req.user.sub}`);
+
+    let a0 = await this.getManagementApi();
+    let profile = await new Promise((resolve, reject) =>
+      a0.getUser(req.user.sub, (err, prof) => err ? reject(err) : resolve(prof)))
```
Comment (jonasfj) [inline src/handlers/mozilla-auth0.js]: ;

Comment (djmitche) [inline src/handlers/mozilla-auth0.js]: what does that mean? :)

File: src/handlers/mozilla-auth0.js (hunk 1 above)
Comment (djmitche) [inline src/handlers/mozilla-auth0.js]: I'm not convinced it's worth embedding retries this low -- if there's an issue, the original API request should retry (or the user should click "login" again).

**Pull request merged**